VirtoCommerce.Storefront/Controllers/AccountController.cs

```diff
@@ -196,6 +197,11 @@ public ActionResult ConfirmInvitation(string organizationId, string email, strin
 
         public async Task<IActionResult> ImpersonateUser(string userId)
         {
+            if (User.Identity.Name == SecurityConstants.AnonymousUsername)
+            {
+                return StoreFrontRedirect($"~/account/login?ReturnUrl={HttpUtility.UrlEncode(Request.Path)}");
```

Comment:
You should avoid of use HttpUtility in asp.net core projects use `IUrlEncoder`  dependency instead. And in this case, encoding absolutely redundant